File: 251001/Ostapchuk/publisher/src/main/java/org/ex/distributed_computing/mapper/PostMapper.java
package org.ex.distributed_computing.mapper;

import org.ex.distributed_computing.dto.request.PostRequestDTO;
import org.ex.distributed_computing.dto.response.PostResponseDTO;
import org.ex.distributed_computing.model.Post;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.util.List;

@Mapper(componentModel = "spring")
public interface PostMapper {

  @Mapping(source = "tweetId", target = "tweet.id")
  Post toEntity(PostRequestDTO dto);

  @Mapping(source = "tweet.id", target = "tweetId")
  PostResponseDTO toDto(Post post);

  List<PostResponseDTO> toDtoList(List<Post> posts);
}

File: 251001/Ostapchuk/discussion/src/main/kotlin/org/ex/discussion/mapper/PostMapper.kt
package org.ex.discussion.mapper

import org.ex.discussion.dto.request.PostRequestDTO
import org.ex.discussion.dto.response.PostResponseDTO
import org.ex.discussion.model.Post
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper
import org.mapstruct.MappingConstants.ComponentModel

@Mapper(
    componentModel = ComponentModel.SPRING,
    injectionStrategy = InjectionStrategy.CONSTRUCTOR
)
interface PostMapper {

    fun toDto(entity: Post): PostResponseDTO

    fun toEntity(dto: PostRequestDTO): Post
}